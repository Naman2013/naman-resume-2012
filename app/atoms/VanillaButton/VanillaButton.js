import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import style from './VanillaButton.style';

const VanillaButton = ({ children, theme, handleClick }) => (
  <Fragment>
    <button onClick={handleClick} style={theme}>
      {children}
    </button>
    <style jsx>{style}</style>
  </Fragment>
);

VanillaButton.propTypes = {
  children: PropTypes.node.isRequired,
  theme: PropTypes.shape({}),
  handleClick: PropTypes.func.isRequired,
};
VanillaButton.defaultProps = { theme: {} };
export default VanillaButton;
