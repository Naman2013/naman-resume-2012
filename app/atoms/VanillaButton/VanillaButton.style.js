import css from 'styled-jsx/css';

export default css`
  button {
    background: none;
    border: none;
    margin: 0;
    padding: 0;
    cursor: pointer;
  }
`;
