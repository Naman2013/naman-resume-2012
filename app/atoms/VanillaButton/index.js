import VanillaButton from './VanillaButton';

export default VanillaButton;
