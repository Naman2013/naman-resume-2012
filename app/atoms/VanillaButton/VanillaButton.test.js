import React from 'react';
import { shallow } from 'enzyme';
import VanillaButton from './VanillaButton';

describe('VanillaButton', () => {
  const shallowWrapper = shallow(<VanillaButton />);
  it('should render correctly', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });
});
