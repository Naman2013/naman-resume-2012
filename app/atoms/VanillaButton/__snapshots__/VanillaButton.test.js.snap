// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VanillaButton should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VanillaButton
    handleClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    theme={Object {}}
  >
    <div />
  </VanillaButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <button
          className="jsx-627253360"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
            }
          }
          style={Object {}}
        >
          <div />
        </button>,
        <JSXStyle
          css={
            Array [
              "button.jsx-627253360{background:none;border:none;margin:0;padding:0;cursor:pointer;-webkit-transition:background-color 0.25s ease-in-out;transition:background-color 0.25s ease-in-out;}",
              "button.jsx-627253360:focus{outline:none;}",
            ]
          }
          styleId="627253360"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div />,
          "className": "jsx-627253360",
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
          },
          "style": Object {},
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            "button.jsx-627253360{background:none;border:none;margin:0;padding:0;cursor:pointer;-webkit-transition:background-color 0.25s ease-in-out;transition:background-color 0.25s ease-in-out;}",
            "button.jsx-627253360:focus{outline:none;}",
          ],
          "styleId": "627253360",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            className="jsx-627253360"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
              }
            }
            style={Object {}}
          >
            <div />
          </button>,
          <JSXStyle
            css={
              Array [
                "button.jsx-627253360{background:none;border:none;margin:0;padding:0;cursor:pointer;-webkit-transition:background-color 0.25s ease-in-out;transition:background-color 0.25s ease-in-out;}",
                "button.jsx-627253360:focus{outline:none;}",
              ]
            }
            styleId="627253360"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div />,
            "className": "jsx-627253360",
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
            },
            "style": Object {},
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              "button.jsx-627253360{background:none;border:none;margin:0;padding:0;cursor:pointer;-webkit-transition:background-color 0.25s ease-in-out;transition:background-color 0.25s ease-in-out;}",
              "button.jsx-627253360:focus{outline:none;}",
            ],
            "styleId": "627253360",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
